src/components/generic/form-autocomplete.tsx

```diff
@@ -0,0 +1,271 @@
+      return (conceptCodes.attributes || [])
+        .filter(attribute => attribute.attributeName && attribute.attributeValue)
+        .reduce((mapped: { [key: string]: string }, attribute: Attribute) => {
+          const key = attribute.attributeName!!;
+          const finalValue = (attribute.attributeValue || []).join(",");
+          mapped[key] = finalValue;
+          return mapped;
+        }, { id: conceptCodes.conceptCodeId!! }) as AutocompleteItem;
+    });
+
+    return autoCompleteItems;
+  };
+  
+  /**
+   * Loads autocomplete items
+   * 
+   * @returns autocomplete items
+   */
+  const loadItems = async () => {
+    const { autocomplete } = field;
+      
+    if (!autocomplete) {
+      throw new Error(autocompleteErrorMessages.MISSING_AUTO_COMPLETE);
+    }
+  
+    switch (autocomplete.service) {
+      case MetaformFieldAutocompleteService.CodeServerConceptCode:
+        return loadCodeServerConceptCodeItems();
+      default:
+        throw new Error(autocompleteErrorMessages.UNKNOWN_AUTO_COMPLETE_SERVICE);
+    }
+  };
+
+  /**
+   * Returns label for autocomplete item
+   * 
+   * @param autocompleteItem autocomplete item
+   * @returns label
+   */
+  const getAutocompleteOptionLabel = (autocompleteItem: AutocompleteItem) => {
+    const { autocomplete } = field;
+
+    if (!autocomplete) {
+      return "Unknown";
+    }
+
+    switch (autocomplete.service) {
+      case MetaformFieldAutocompleteService.CodeServerConceptCode:
+        return autocompleteItem.Abbreviation;
+      default:
+        return "Unknown";
+    }
+  };
+
+  /**
+   * Returns fields that use this autocomplete field as source
+   * 
+   * @returns fields that use this autocomplete field as source
+   */
+  const getSourceFields = () => {
+    return (metaform.sections || [])
+      .flatMap(section => section.fields || [])
+      .filter(sourceField => sourceField.source?.type === MetaformFieldSourceType.Autocomplete)
+      .filter(sourceField => sourceField.source?.options?.autocompleteField === field.name);
+  };
+
+  /**
+   * Loading items
+   */
+  const loadingItems = async () => {
+    setLoading(true);
+
+    try {
+      const loadedItems = await loadItems();
+      const defaultAutoCompleteItem = loadedItems.find(item => item.id === value as string);
+
+      setItems(loadedItems);
+      setLoading(false);
+      setDefaultValue(defaultAutoCompleteItem ? { ...defaultAutoCompleteItem } : defaultAutoCompleteItem);
+
+      getSourceFields().forEach(sourceField => {
+        if (sourceField.name) {
+          const itemProperty = sourceField.source?.options?.autocompleteItemProperty;
+          const itemValue = itemProperty && defaultAutoCompleteItem ? defaultAutoCompleteItem[itemProperty] : null;
+          setFieldValue(sourceField.name, itemValue);
+        }
+      });
+    } catch (e: any) {
+      setLoading(false);
+      setErrorMessage(e.message);
```

Comment:
Could the errorContext be used here to display error?

Reply:
I'll probably leave this as is for now as the proper error handler will be implemented by Eetu and I think it might have been something he suggested as well.